plugins/woocommerce/src/StoreApi/docs/checkout.md

```diff
@@ -23,48 +23,128 @@ There are no parameters required for this endpoint.
+    "address_1": "550 Central Park West",
+    "address_2": "Corner Penthouse Spook Central",
+    "city": "New York",
+    "state": "NY",
+    "postcode": "10023",
+    "country": "US"
+  },
+  "payment_method": "",
+  "payment_result": {
+    "payment_status": "",
+    "payment_details": [],
+    "redirect_url": ""
+  }
 }
 ```
 
+## Update checkout data
+
+This endpoint allows you to update the checkout data for the current order. This can be called from the frontend to persist checkout fields, for example.
+
+```http
+PUT /wc/store/v1/checkout?__experimental_calc_totals=true
+```
+
+Note the `__experimental_calc_totals` parameter. This is used to determine if the cart totals should be recalculated. This should be set to true if the cart totals are being updated in response to a PUT request, false otherwise.
+
+| Attribute           | Type   | Required | Description                                         |
+| :------------------ | :----- | :------: | :-------------------------------------------------- |
+| `additional_fields` | object |    No    | Name => value pairs of additional fields to update. |
+| `payment_method`    | string |    No    | The ID of the payment method selected.              |
+| `order_notes`       | string |    No    | Order notes.                                        |
+
+```sh
+curl --header "Nonce: 12345" --request PUT https://example-store.com/wp-json/wc/store/v1/checkout?additional_fields[plugin-namespace/leave-on-porch]=true&additional_fields[plugin-namespace/location-on-porch]=dsdd&payment_method=bacs&order_notes=Please%20leave%20package%20on%20back%20porch
+```
+
+### Example Request
+
+```json
+{
+  "additional_fields": {
+    "plugin-namespace/leave-on-porch": true,
+    "plugin-namespace/location-on-porch": "dsdd"
+  },
+  "payment_method": "bacs",
+  "order_notes": "Please leave package on back porch"
+}
+```
+
+### Example Response
+
+```json
+{
+    "order_id": 1486,
+    "status": "checkout-draft",
+    "order_key": "wc_order_KLpMaJ054PVlb",
+    "order_number": "1486",
+    "customer_note": "",
+    "customer_id": 1,
+    "billing_address": {
+			"first_name": "Peter",
+			"last_name": "Venkman",
+			"company": "",
+			"address_1": "550 Central Park West",
+			"address_2": "Corner Penthouse Spook Central",
+			"city": "New York",
+			"state": "NY",
+			"postcode": "10023",
+			"country": "US",
+			"email": "admin@example.com",
+			"phone": "555-2368"
+		},
+		"shipping_address": {
+			"first_name": "Peter",
+			"last_name": "Venkman",
+			"company": "",
+			"address_1": "550 Central Park West",
+			"address_2": "Corner Penthouse Spook Central",
+			"city": "New York",
+			"state": "NY",
+			"postcode": "10023",
+			"country": "US"
+		},
+    "payment_method": "bacs",
+    "payment_result": null,
+    "additional_fields": {
+        "plugin-namespace/leave-on-porch": true,
+        "plugin-namespace/location-on-porch": "dsdd"
+    },
+    "__experimentalCart": { ... },
+    "extensions": {}
+}
+```
```

Comment:
Lots of other lines were updated to use spaces, this is using a mixture of tabs and spaces. Let's update it to be consistent with the rest of the file.